Mathlib/RingTheory/Flat.lean

```diff
@@ -76,6 +99,100 @@ instance self (R : Type u) [CommRing R] : Flat R R :=
       lift.tmul, Submodule.subtype_apply, Algebra.id.smul_eq_mul, lsmul_apply]⟩
 #align module.flat.self Module.Flat.self
 
+
+/-- A retract of a flat `R`-module is flat. -/
+lemma of_retract (R : Type u) (M : Type v) (N : Type w)
+    [CommRing R] [AddCommGroup M] [AddCommGroup N] [Module R M] [Module R N]
+    (i : N →ₗ[R] M) (r : M →ₗ[R] N) (h : r.comp i = LinearMap.id)
+    (f : Flat R M) : Flat R N := by
```

Comment:
Is there a reason why `f` is not an instance argument? (same below)

Reply:
No particular reason. I made `f` an instance argument in more cases now.